src/main/java/org/embulk/input/zendesk/services/ZendeskNormalServices.java

```diff
@@ -0,0 +1,239 @@
+    {
+        if (path.isEmpty()) {
+            path = buildURI(page, startTime);
+        }
+
+        final String response = getZendeskRestClient().doGet(path, task, isPreview);
+        return ZendeskUtils.parseJsonObject(response);
+    }
+
+    protected abstract String buildURI(int page, long startTime);
+
+    @VisibleForTesting
+    protected ZendeskRestClient getZendeskRestClient()
+    {
+        if (zendeskRestClient == null) {
+            zendeskRestClient = new ZendeskRestClient();
+        }
+        return zendeskRestClient;
+    }
+
+    private void importDataForIncremental(final ZendeskInputPlugin.PluginTask task, final RecordImporter recordImporter, final TaskReport taskReport)
+    {
+        long startTime = 0;
+
+        if (task.getStartTime().isPresent()) {
+            startTime = ZendeskDateUtils.getStartTime(task.getStartTime().get());
+        }
+
+        // For incremental target, we will run in one task but split in multiple threads inside for data deduplication.
+        // Run with incremental will contain duplicated data.
+        ThreadPoolExecutor pool = null;
+        try {
+            final Set<String> knownIds = ConcurrentHashMap.newKeySet();
+            pool = new ThreadPoolExecutor(
+                    10, 100, 60, TimeUnit.SECONDS, new LinkedBlockingQueue<>()
+            );
+
+            while (true) {
+                int recordCount = 0;
+
+                // Page argument isn't used in incremental API so we just set it to 0
+                final JsonNode result = getData("", 0, false, startTime);
+                final Iterator<JsonNode> iterator = ZendeskUtils.getListRecords(result, task.getTarget().getJsonName());
+
+                int numberOfRecords = 0;
+                if (result.has(ZendeskConstants.Field.COUNT)) {
+                    numberOfRecords = result.get(ZendeskConstants.Field.COUNT).asInt();
+                }
+
+                while (iterator.hasNext()) {
+                    final JsonNode recordJsonNode = iterator.next();
+
+                    if (isUpdatedBySystem(recordJsonNode, startTime)) {
+                        continue;
+                    }
+
+                    if (task.getDedup()) {
+                        final String recordID = recordJsonNode.get(ZendeskConstants.Field.ID).asText();
+
+                        // add success -> no duplicate
+                        if (!knownIds.add(recordID)) {
+                            continue;
+                        }
+                    }
+
+                    pool.submit(() -> fetchRecord(recordJsonNode, task, recordImporter));
+                    recordCount++;
+                    if (Exec.isPreview()) {
+                        return;
+                    }
+                }
+
+                logger.info("Fetched '{}' records from start_time '{}'", recordCount, startTime);
+
+                startTime = result.get(ZendeskConstants.Field.END_TIME).asLong();
```

Comment:
It could be an edge case. 
More information: https://develop.zendesk.com/hc/en-us/community/posts/360001646028-Incremental-Ticket-Export-more-than-1000-updates-per-second